plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java

```diff
@@ -823,34 +834,62 @@ private String toRemoteSchemaName(String schemaName)
         if (!caseInsensitiveNameMatching) {
             return schemaName;
         }
-        for (String remoteSchemaName : client.listDatabaseNames()) {
+        for (String remoteSchemaName : listDatabaseNames()) {
             if (schemaName.equals(remoteSchemaName.toLowerCase(ENGLISH))) {
                 return remoteSchemaName;
             }
         }
         return schemaName;
     }
 
+    private Iterable<String> listDatabaseNames()
+    {
+        OperationExecutor operationExecutor = ((MongoClientImpl) client).getOperationExecutor();
+        CodecRegistry codecRegistry = ((MongoClientImpl) client).getSettings().getCodecRegistry();
+        boolean retryReads = ((MongoClientImpl) client).getSettings().getRetryReads();
+        return new ListDatabasesIterableImpl<>(null, BsonDocument.class, codecRegistry, ReadPreference.primary(), operationExecutor, retryReads)
```

Comment:
Any reason on why are we extracting this here ?

Reply:
Can we use `listDatabases` and apply the changes on top of them ?

Reply:
Sorry I do not understand what you mean. Could you explain more?

Reply:
Oh, I did not notice this before, I will change it

Reply:
Updated